src/uwtools/cli.py

```diff
@@ -269,6 +272,99 @@ def _dispatch_config_validate(args: Args) -> bool:
     return uwtools.api.config.validate(schema_file=args[STR.schemafile], config=args[STR.infile])
 
 
+# Mode file
+
+
+def _add_subparser_file(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: file
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.file, "Handle files")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.action, STR.action.upper())
+    return {
+        STR.copy: _add_subparser_file_copy(subparsers),
+        STR.link: _add_subparser_file_link(subparsers),
+    }
+
+
+def _add_subparser_file_copy(subparsers: Subparsers) -> ActionChecks:
+    """
+    Subparser for mode: file copy
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.copy, "Copy files")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_target_dir(required, required=True)
+    optional = _basic_setup(parser)
+    _add_arg_config_file(group=optional, required=False)
+    _add_arg_dry_run(optional)
+    checks = _add_args_verbosity(optional)
+    _add_arg_keys(optional)
+    return checks
+
+
+def _add_subparser_file_link(subparsers: Subparsers) -> ActionChecks:
+    """
+    Subparser for mode: file link
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.link, "Link files")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_target_dir(required, required=True)
+    optional = _basic_setup(parser)
+    _add_arg_config_file(group=optional, required=False)
+    _add_arg_dry_run(optional)
+    checks = _add_args_verbosity(optional)
+    _add_arg_keys(optional)
+    return checks
```

Comment:
The previous two functions feel pretty WET.

Reply:
I'll DRY those out a bit.